fdbserver/VersionedBTree.actor.cpp

```diff
@@ -8030,7 +8030,8 @@ class KeyValueStoreRedwood : public IKeyValueStore {
 				// Run the destructive sanity check, but don't throw.
 				ErrorOr<Void> err = wait(errorOr(self->m_tree->clearAllAndCheckSanity()));
 				// If the test threw an error, it must be an injected fault or something has gone wrong.
-				ASSERT(!err.isError() || err.getError().isInjectedFault());
+				ASSERT(!err.isError() || err.getError().isInjectedFault() ||
```

Comment:
Is there a reason not to make `unexpected_encoding_type` an injected fault?

Reply:
because it is not an injected fault. An injected fault is one that injected by, for example, the MachineAttrition workload. The `unexpected_encoding_type` is a normal error. The unit test being fix doesn't inject the error, but just try to trigger it.